x-pack/plugin/security/qa/multi-cluster/src/javaRestTest/java/org/elasticsearch/xpack/remotecluster/RemoteClusterSecurityTransformIT.java

```diff
@@ -0,0 +1,196 @@
+            .setting("xpack.security.authc.token.enabled", "true")
+            .setting("logger.org.elasticsearch.xpack.security.transport.CrossClusterAccessServerTransportFilter", "debug")
+            .keystore("xpack.security.remote_cluster_server.ssl.secure_key_passphrase", "remote-cluster-password")
+            .build();
+
+        queryCluster = ElasticsearchCluster.local()
+            .name("query-cluster")
+            .apply(commonClusterConfig)
+            .module("transform")
+            .module("reindex")
+            .setting("logger.org.elasticsearch.xpack.security.authz.AuthorizationService", "debug")
+            .setting("xpack.security.remote_cluster_client.ssl.enabled", "true")
+            .setting("xpack.security.remote_cluster_client.ssl.certificate_authorities", "remote-cluster-ca.crt")
+            .setting("xpack.security.authc.token.enabled", "true")
+            .keystore("cluster.remote.my_remote_cluster.credentials", () -> {
+                API_KEY_MAP_REF.compareAndSet(null, createCrossClusterAccessApiKey("""
+                    [
+                      {
+                         "names": ["shared-transform-index"],
+                         "privileges": ["read", "read_cross_cluster", "view_index_metadata"]
+                      }
+                    ]"""));
+                return (String) API_KEY_MAP_REF.get().get("encoded");
+            })
+            .rolesFile(Resource.fromClasspath("roles.yml"))
+            .user(REMOTE_TRANSFORM_USER, PASS.toString(), "transform_admin,transform_remote_shared_index")
+            .build();
+    }
+
+    @ClassRule
+    // Use a RuleChain to ensure that fulfilling cluster is started before query cluster
+    public static TestRule clusterRule = RuleChain.outerRule(fulfillingCluster).around(queryCluster);
+
+    public void testCrossClusterTransform() throws Exception {
+        configureRemoteClusters();
+
+        // Fulfilling cluster
+        {
+            final Request createIndexRequest = new Request("PUT", "shared-transform-index");
+            createIndexRequest.setJsonEntity("""
+                {
+                  "mappings": {
+                    "properties": {
+                      "time": { "type": "date" },
+                      "user": { "type": "keyword" },
+                      "stars": { "type": "integer" },
+                      "coolness": { "type": "integer" }
+                    }
+                  }
+                }""");
+            assertOK(performRequestAgainstFulfillingCluster(createIndexRequest));
+
+            // Index some documents, so we can attempt to transform them from the querying cluster
+            final Request bulkRequest = new Request("POST", "/_bulk?refresh=true");
+            bulkRequest.setJsonEntity(Strings.format("""
+                {"index": {"_index": "shared-transform-index"}}
+                {"user": "a", "stars": 1, "date" : "2018-10-29T12:12:12.123456789Z"}
```

Comment:
The `date` field isn't really relevant to the test, right? If so, a nit: wondering if we should omit it to keep the tests more focused.

Reply:
No I don't think it is necessary. The test setup is copied from https://github.com/elastic/elasticsearch/blob/04aadd7870d734e2c0e13dd5b59fc8b4990317fa/x-pack/plugin/transform/qa/multi-cluster-tests-with-security/src/test/resources/rest-api-spec/test/remote_cluster/80_transform.yml#L64-L65

I'll clean it up here.